Issue: Explicit type annotation is needed when initialising graphQL tada

### Describe the bug

build fails with typescript v5.5 onwards while having the option "declaration" enabled in tsconfig.
```ts
// tsconfig.json

{
  "compilerOptions": {
    "declaration": true,
    ...
  }
}
```

```ts
// graphql.ts

import { initGraphQLTada } from 'gql.tada';
import type { introspection } from './generated/supergraph.d.ts';

export const graphql = initGraphQLTada<{
  disableMasking: true;
  introspection: introspection;
}>();

```

raised error:
```
error TS7056: The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.

export const graphql = initGraphQLTada<{
```


### Reproduction

_No response_

### gql.tada version

gql.tada v1.8.6

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/0no-co/gql.tada/discussions) should be used
- [X] Read the [docs](https://gql-tada.0no.co/).
- [X] Follow our [Code of Conduct](https://github.com/0no-co/gql.tada/blob/main/CODE_OF_CONDUCT.md)

--- a/.changeset/empty-hairs-punch.md
+++ b/.changeset/empty-hairs-punch.md
@@ -1,0 +1,5 @@
+---
+'gql.tada': patch
+---
+
+Update internal type signature of `initGraphQLTada<>()` function and include `initGraphQLTada<Setup>` alias type. This alias type makes it more intuitive to declare types the return type of `initGraphQLTada()`, since it mirrors the same name.

--- a/src/api.ts
+++ b/src/api.ts
@@ -259,6 +259,33 @@ type configOfSetup<Setup extends AbstractSetupSchema> = {
   isMaskingDisabled: Setup['disableMasking'] extends true ? true : false;
 };
 
+/** Utility to type-instantiate a `graphql` document function with.
+ *
+ * @remarks
+ * The `initGraphQLTada` type may be used to manually instantiate a type
+ * as returned by `initGraphQLTada<>()` with the same input type.
+ *
+ * @example
+ * ```
+ * import { initGraphQLTada } from 'gql.tada';
+ * import type { myIntrospection } from './myIntrospection';
+ *
+ * export const graphql: initGraphQLTada<{
+ *   introspection: typeof myIntrospection;
+ *   scalars: {
+ *     DateTime: string;
+ *     Json: any;
+ *   };
+ * }> = initGraphQLTada();
+ *
+ * const query = graphql(`{ __typename }`);
+ * ```
+ */
+export type initGraphQLTada<Setup extends AbstractSetupSchema> = GraphQLTadaAPI<
+  schemaOfSetup<Setup>,
+  configOfSetup<Setup>
+>;
+
 /** Setup function to create a typed `graphql` document function with.
  *
  * @remarks
@@ -285,7 +312,7 @@ type configOfSetup<Setup extends AbstractSetupSchema> = {
  * const query = graphql(`{ __typename }`);
  * ```
  */
-function initGraphQLTada<const Setup extends AbstractSetupSchema>() {
+export function initGraphQLTada<const Setup extends AbstractSetupSchema>(): initGraphQLTada<Setup> {
   type Schema = schemaOfSetup<Setup>;
   type Config = configOfSetup<Setup>;
 
@@ -709,12 +736,9 @@ function unsafe_readResult<
   return data as any;
 }
 
-const graphql: GraphQLTadaAPI<
-  schemaOfSetup<setupSchema>,
-  configOfSetup<setupSchema>
-> = initGraphQLTada();
+const graphql: initGraphQLTada<setupSchema> = initGraphQLTada();
 
-export { parse, graphql, readFragment, maskFragments, unsafe_readResult, initGraphQLTada };
+export { parse, graphql, readFragment, maskFragments, unsafe_readResult };
 
 export type {
   setupCache,

